crates/host/src/wasmbus/mod.rs

```diff
@@ -2000,13 +1993,84 @@ impl Host {
                 .context("failed to write newline")?;
             stdin.shutdown().await.context("failed to close stdin")?;
 
+            let prov_nats = self.prov_rpc_nats.clone();
+            let ctl_nats = self.ctl_nats.clone();
+            let event_builder = self.event_builder.clone();
+            // NOTE: health_ prefix here is to allow us to move the variables into the closure
+            let health_lattice_prefix = self.host_config.lattice_prefix.clone();
+            let health_provider_id = claims.subject.to_string();
+            let health_link_name = link_name.to_string();
+            let health_contract_id = claims.metadata.clone().map(|m| m.capid).unwrap_or_default();
```

Comment:
Nothing necessarily needs to change here, but just want to point out that we could easily just change method receiver to `Arc<Self>` and then clone the Arc of the host itself and move it into the closure.

Note that it would make `event::publish` redundant and remove a bunch of clones here